lib/iex/lib/iex/broker.ex

```diff
@@ -0,0 +1,176 @@
+defmodule IEx.Broker do
+  @moduledoc false
+  @name IEx.Broker
+
+  @type take_ref :: {takeover_ref :: reference(), server_ref :: reference()}
+
+  use GenServer
+
+  @doc """
+  Finds the IEx server running inside `:user_drv`, on this node exclusively.
+  """
+  @spec shell :: pid | nil
+  def shell() do
+    # Locate top group leader, always registered as user
+    # can be implemented by group (normally) or user
+    # (if oldshell or noshell)
+    if user = Process.whereis(:user) do
```

Comment:
do we need an if here? Could we just match on the user and fail?

Reply:
We need this because we can't pass user as nil later on. So we either need an `if` or pattern matching.